atef/widgets/config/run_base.py

```diff
@@ -275,7 +340,8 @@ def setup_buttons(self, configs, next_widget: AtefItem = None) -> None:
 
     def _make_run_slot(self, configs) -> None:
 
-        def run_slot():
+        @busy_cursor
```

Comment:
`BusyCursorThread` looks neat! You might want a `try/finally` block around it (along with an `except Exception: raised_exception.emit(ex)` so you can catch it if failed by way of a signal)

I think the desire for wanting blocking behavior is understandable. It just needs to be implemented by other means (that is to say, blocking the GUI thread busily doing non-GUI tasks is truly never the answer).  It may be sensible to disable the specific file pane when such things are running, but you wouldn't want it to interrupt access to other files you have open.

With this running in a background thread now, you could also have a (pretty crucial) hook for stopping checkout tasks that are taking too long for any reason. That'll require some more care/attention to detail, but backgrounding it is I think the first pass at it.

It's a good point about the lack of put completion for PV-only signals. It may be worth adding in the concept of a setpoint PV and a readback PV... For better or worse, that's an essential part of EPICS V3.